hw3/01-routing.js

```diff
@@ -43,7 +46,67 @@ const server = http.createServer((req, res) => {
     res.end();
   }
 
-  // Add your code here
+  else if (req.url === '/welcome') {
+    let routeResults = getRoutes();
+
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>Welcome to HW #3!</h1>`);
+    res.write(`<ul> ${routeResults} </ul>`);
+    res.end();
+  }
+
+  else if (req.url === '/redirect') {
+    res.writeHead(302, { Location: "/redirected" });
+    res.end();
+  }
+
+  else if (req.url === '/redirected') {
+    let routeResults = getRoutes();
+
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write(`<h1>Redirected to /redirected</h1>`);
+    res.write(`<ul> ${routeResults} </ul>`);
+    res.end();
+  }
+
+  else if (req.url === '/cache') {
+    let routeResults = getRoutes();
+
+    res.writeHead(200, { 'Content-Type': 'text/html', 'Cache-control': 'public, max-age=86400'});
+    res.write(`<h1>This resource was cached</h1>`);
+    res.write(`<ul> ${routeResults} </ul>`);
+    res.end();
+  }
+
+  else if (req.url === '/cookie') {
+    let routeResults = getRoutes();
+
+    res.writeHead(200, { 'Content-Type': 'text/html', 'Set-Cookie':'hello=world'});
+    res.write(`<h1>cookies... yummm</h1>`);
+    res.write(`<ul> ${routeResults} </ul>`);
+    res.end();
+  }
+
+  else if (req.url === '/check-cookies') {
+    let routeResults = getRoutes();
```

Comment:
routeResults is never used in the check-cookies route. 
Everything in 01 looks great.
Good job.